client/rest/src/main/java/org/opensearch/client/RestClient.java

```diff
@@ -674,20 +674,18 @@ private HttpUriRequestBase addRequestBody(HttpUriRequestBase httpRequest, HttpEn
     static URI buildUri(String pathPrefix, String path, Map<String, String> params) {
         Objects.requireNonNull(path, "path must not be null");
         try {
-            String fullPath;
-            if (pathPrefix != null && pathPrefix.isEmpty() == false) {
-                if (pathPrefix.endsWith("/") && path.startsWith("/")) {
-                    fullPath = pathPrefix.substring(0, pathPrefix.length() - 1) + path;
-                } else if (pathPrefix.endsWith("/") || path.startsWith("/")) {
-                    fullPath = pathPrefix + path;
-                } else {
-                    fullPath = pathPrefix + "/" + path;
-                }
-            } else {
-                fullPath = path;
+            URIBuilder uriBuilder = new URIBuilder();
+
+            if (pathPrefix != null && !pathPrefix.isEmpty() && !"/".equals(pathPrefix)) {
```

Comment:
I am not sure the equality checks are suitable here, fe passing "//"  as path would be a miss, the previous code definitely was more robust towards such cases.

Reply:
In this case I was skipping a single `/` due to the logic of `URIBuilder.appendPath`:
```
new URIBuilder().appendPath("/").appendPath("/") -> //
new URIBuilder().appendPath("/foo").appendPath("/bar") -> /foo/bar
```

I've combined this `URIBuilder.appendPath` approach with the previous `trimSlashes` implementation and some extra test cases to see how we feel about this level of normalization. Though this will strip duplicate `/` in leading or trailing position, it won't touch `foo////bar`. If we would want to normalize that, then this could be changed to instead split on `/` and skip empty segments.